### PR TITLE
Make the loading interstitials a bit nicer

### PR DESCRIPTION
**Pull request by bglw**

### CODE REVIEW

File: truncate_client/src/app_inner.rs [hunk 1]
```diff
@@ -141,29 +144,93 @@ pub fn render(client: &mut OuterApplication, ui: &mut egui::Ui) {
...
+                    #[cfg(target_arch = "wasm32")]
+                    {
+                        web_sys::window().unwrap().location().reload();
+                    }
+                }
+            });
         }
         GameStatus::PendingCreate => {
-            ui.label("Waiting for a new game to be created . . .");
+            let dot_count = (current_time.as_millis() / 500) % 4;
```
Comment (BlakeMScurr) [inline truncate_client/src/app_inner.rs:189]: Code between `PendingJoin` and `PendingCreate` is basically duplicated yeah? Nbd though

File: truncate_client/src/app_outer.rs [hunk 2]
```diff
@@ -127,6 +127,16 @@ fn load_map_texture(ctx: &egui::Context) -> TextureHandle {
...
     fn update(&mut self, ctx: &egui::Context, _frame: &mut eframe::Frame) {
+        // We have to go through the instant crate as
+        // most std time functions are not implemented
+        // in Rust's wasm targets.
+        // instant::SystemTime::now() conditionally uses
+        // a js function on wasm targets, and otherwise aliases
+        // to the std SystemTime type.
+        let current_time = instant::SystemTime::now()
+            .duration_since(instant::SystemTime::UNIX_EPOCH)
+            .expect("Please don't play Truncate earlier than 1970");
```
Comment (BlakeMScurr) [inline truncate_client/src/app_outer.rs:138]: 🕺

**Review by BlakeMScurr — APPROVED** [review]
LGTM

File: truncate_client/src/app_inner.rs (hunk 1 above)
Comment (bglw) [reply truncate_client/src/app_inner.rs:189]: For now yeah. I'd like to move it all the way out to be some nice "dottable" component or something but not right now

**Review by BlakeMScurr — APPROVED** [review]

**Pull request merged**